### DISCUSSION TITLE
`isDirty` on the form helper?

### DISCUSSION THREAD

**Discussion by LucaRed** [inertiajs/inertia#646] [Ideas]
Is there any way to check if the current form has been modified (compared to its initial state)?

Something like the [`isDirty()` Laravel method](https://laravel.com/docs/8.x/eloquent#examining-attribute-changes) or [`ActiveModel::Dirty` on Rails](https://api.rubyonrails.org/classes/ActiveModel/Dirty.html).

My use case: I'd like to check if the form has been modified in order to ask for confirmation ("Are you sure you want to leave the page?") on the `onbeforeunload` event.

**Comment by pikant** [comment]
As mentioned here (https://github.com/inertiajs/inertia/discussions/653) I think this would be a very helpful feature as well.

**Reply by giffeler** [reply]
Good idea.

**Reply by thkrueger** [reply]
Would be a great feature!

**Comment by plakhin** [comment]
Since there's already `defaults` const there it shouldn't be hard to compare two objects, something like `JSON.stringify(this) ===  JSON.stringify(defaults)` can be used.

**Comment by reinink** [comment]
I've started a new PR for this feature: https://github.com/inertiajs/inertia/pull/660
